storage/innobase/handler/handler0alter.cc

```diff
@@ -4342,10 +4418,21 @@ innobase_check_foreigns(
 				return field.field == *fp;
 			});
 
-		if (it == end || (it->flags & NOT_NULL_FLAG)) {
-			if (innobase_check_foreigns_low(
+		if (it == end) {
+			if (innobase_check_foreign_drop_col(
+				user_table, drop_fk, n_drop_fk,
+				(*fp)->field_name.str)) {
+				return true;
+			}
+			continue;
+		}
+
+
+		if ((it->flags & NOT_NULL_FLAG)
```

Comment:
You could remove the `continue` and add `else` before the `if` to make this slightly more readable.